libsolidity/ast/Types.cpp

```diff
@@ -2034,15 +2036,16 @@ bool StructType::operator==(Type const& _other) const
 	return ReferenceType::operator==(other) && other.m_struct == m_struct;
 }
 
-unsigned StructType::calldataEncodedSize(bool _padded) const
+unsigned StructType::calldataEncodedSize(bool) const
 {
 	unsigned size = 0;
 	for (auto const& member: members(nullptr))
 		if (!member.type->canLiveOutsideStorage())
 			return 0;
 		else
 		{
-			unsigned memberSize = member.type->calldataEncodedSize(_padded);
```

Comment:
This change is hard to unit-test, because you would need to construct an AST. I will test it with #5812 .